### PR TITLE
Add remote and local addresses to GCDWebServerRequest

### PR DESCRIPTION
**Pull request by sergiou87**
Coming from #152, sorry it took this long for me to implement it :sweat_smile:

### CODE REVIEW

File: GCDWebServer/Core/GCDWebServerConnection.m [hunk 1]
```diff
@@ -548,6 +548,8 @@ - (void)_readRequestHeaders {
           }
         }
         if (_request) {
+          _request.localAddressData = [self.localAddressData copy];
+          _request.remoteAddressData = [self.remoteAddressData copy];
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerConnection.m]: Copy is not needed and it were it would have to be done by property setter not the caller / user of the property.

File: GCDWebServer/Core/GCDWebServerRequest.m [hunk 2]
```diff
@@ -308,6 +310,14 @@ - (void)setAttribute:(id)attribute forKey:(NSString*)key {
   [_attributes setValue:attribute forKey:key];
 }
 
+- (NSString*)localAddressString {
+    return GCDWebServerStringFromSockAddr(_localAddress.bytes, YES);
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerRequest.m]: whitespace

File: GCDWebServer/Core/GCDWebServerRequest.m [hunk 3]
```diff
@@ -308,6 +310,14 @@ - (void)setAttribute:(id)attribute forKey:(NSString*)key {
...
+}
+
+- (NSString*)remoteAddressString {
+    return GCDWebServerStringFromSockAddr(_remoteAddress.bytes, YES);
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerRequest.m]: whitespace

File: GCDWebServer/Core/GCDWebServerRequest.m [hunk 4]
```diff
@@ -157,7 +157,9 @@ @interface GCDWebServerRequest () {
   NSString* _noneMatch;
   NSRange _range;
   BOOL _gzipAccepted;
-  
+  NSData* _localAddress;
+  NSData* _remoteAddress;
+
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerRequest.m]: unnecessary whitespace diff

File: GCDWebServer/Core/GCDWebServerRequest.h [hunk 5]
```diff
@@ -158,6 +158,30 @@ extern NSString* const GCDWebServerRequestAttribute_RegexCaptures;
 @property(nonatomic, readonly) BOOL acceptsGzipContentEncoding;
 
 /**
```
Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerRequest.h]: You copy pasted without updating the doc: "of the connection" -> "for the request"

File: GCDWebServer/Core/GCDWebServerConnection.m (hunk 1 above)
Comment (sergiou87) [inline GCDWebServer/Core/GCDWebServerConnection.m]: Agree, changed!

File: GCDWebServer/Core/GCDWebServerRequest.h (hunk 5 above)
Comment (sergiou87) [inline GCDWebServer/Core/GCDWebServerRequest.h]: Right, updating…

**Comment by sergiou87** [comment]
Anything else? :smile:

**Comment by swisspol** [comment]
You still have one comment unaddressed above :)

File: GCDWebServer/Core/GCDWebServerRequest.m (hunk 4 above)
Comment (sergiou87) [inline GCDWebServer/Core/GCDWebServerRequest.m]: Sorry I missed this one :sweat_smile: 

What's the problem with it exactly? I mean, the the empty line was there before, so is it just that or is there something I'm missing?

Comment (swisspol) [inline GCDWebServer/Core/GCDWebServerRequest.m]: You changed the whitespace.

**Comment by sergiou87** [comment]
Done, I think :smile:

**Comment by swisspol** [comment]
Looks good, thx

**Comment by sergiou87** [comment]
Thank you!!
